src/main/kotlin/org/jetbrains/research/ml/codetracker/server/TrackerQueryExecutor.kt

```diff
@@ -1,72 +1,130 @@
 package org.jetbrains.research.ml.codetracker.server
 
-import org.jetbrains.research.ml.codetracker.*
 import com.intellij.openapi.application.PathManager
-import io.reactivex.internal.operators.maybe.MaybeDoAfterSuccess
-import org.jetbrains.research.ml.codetracker.models.Extension
 import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
 import okhttp3.MultipartBody
 import okhttp3.Request
 import okhttp3.RequestBody.Companion.asRequestBody
 import okhttp3.RequestBody.Companion.toRequestBody
-import okhttp3.Response
+import org.jetbrains.research.ml.codetracker.Plugin
+import org.jetbrains.research.ml.codetracker.models.Extension
+import org.jetbrains.research.ml.codetracker.tracking.StoredInfoWrapper
 import java.io.File
-import java.lang.IllegalStateException
+import java.io.PrintWriter
 import java.net.URL
 
+
 object TrackerQueryExecutor : QueryExecutor() {
 
     private val ACTIVITY_TRACKER_FILE = "ide-events${Extension.CSV.ext}"
     private const val CODE_TRACKER_FILE_FIELD = "codetracker"
     private const val ACTIVITY_TRACKER_FILE_FIELD = "activitytracker"
-    private val activityTrackerPath = "${PathManager.getPluginsPath()}/activity-tracker/" + ACTIVITY_TRACKER_FILE
+    private val activityTrackerPath = "${PathManager.getPluginsPath()}/activity-tracker/$ACTIVITY_TRACKER_FILE"
+    private const val DEFAULT_ACTIVITY_TRACKER_ID = "-1"
 
-    private var activityTrackerKey: String? = null
+    var userId: String? = null
 
     init {
-        initActivityTrackerInfo()
+        StoredInfoWrapper.info.userId?.let {
+            userId = it
+        } ?: run {
+            initUserId()
+            StoredInfoWrapper.updateStoredInfo(userId = userId)
+        }
```

Comment:
Why is it a new code? We gathered data with this feature. Maybe should we merge master into dev? Or am I confusing something?